nbgitpuller/handlers.py

```diff
@@ -29,6 +29,12 @@ def __init__(self, *args, **kwargs):
         if 'git_lock' not in self.settings:
             self.settings['git_lock'] = locks.Lock()
 
+    def get_login_url(self):
+        # raise on failed auth, not redirect
+        # can't redirect EventStream to login
+        # same as Jupyter's APIHandler
+        raise web.HTTPError(403)
```

Comment:
An alternative would be for the EventStream to inherit from APIHandler instead of the base JupyterHandler, which already has this logic.